internal/service/arcregionswitch/plan.go

```diff
@@ -0,0 +1,1606 @@
+	RecoveryApproach             fwtypes.StringEnum[awstypes.RecoveryApproach]        `tfsdk:"recovery_approach"`
+	Regions                      fwtypes.ListOfString                                 `tfsdk:"regions"`
+	Description                  types.String                                         `tfsdk:"description"`
+	PrimaryRegion                types.String                                         `tfsdk:"primary_region"`
+	RecoveryTimeObjectiveMinutes types.Int64                                          `tfsdk:"recovery_time_objective_minutes"`
+	AssociatedAlarms             fwtypes.SetNestedObjectValueOf[associatedAlarmModel] `tfsdk:"associated_alarms"`
+	Workflows                    fwtypes.ListNestedObjectValueOf[workflowModel]       `tfsdk:"workflow"`
+	Triggers                     fwtypes.ListNestedObjectValueOf[triggerModel]        `tfsdk:"triggers"`
+	Region                       types.String                                         `tfsdk:"region"`
+	Tags                         tftags.Map                                           `tfsdk:"tags"`
+	TagsAll                      tftags.Map                                           `tfsdk:"tags_all"`
+}
+
+// Custom expand to handle AssociatedAlarms set-to-map conversion
+// Expand converts the Terraform resource model to AWS API input.
+// Custom expansion is required because:
+// 1. Union Type Handling: ExecutionBlockConfiguration uses AWS SDK union types that AutoFlex cannot handle automatically
+// 2. Complex Nested Transformations: ScalingResources (list→map[string]map[string]) and RegionAndRoutingControls (list→map[string][]) require manual conversion
+// 3. Set-to-Map Conversions: AssociatedAlarms transforms from Terraform sets to AWS API maps with complex key generation
+// 4. Conditional Logic: Different execution block types require different field mappings and validations
+// AutoFlex works well for simple field mappings but cannot handle these complex structural transformations.
+func (m resourcePlanModel) Expand(ctx context.Context) (result any, diags fwdiag.Diagnostics) {
+	var apiObject arcregionswitch.CreatePlanInput
+
+	// Manually expand all fields except AssociatedAlarms and complex nested structures
+	diags.Append(flex.Expand(ctx, m.Name, &apiObject.Name)...)
+	diags.Append(flex.Expand(ctx, m.ExecutionRole, &apiObject.ExecutionRole)...)
+	diags.Append(flex.Expand(ctx, m.RecoveryApproach, &apiObject.RecoveryApproach)...)
+	diags.Append(flex.Expand(ctx, m.Regions, &apiObject.Regions)...)
+	diags.Append(flex.Expand(ctx, m.Description, &apiObject.Description)...)
+	diags.Append(flex.Expand(ctx, m.PrimaryRegion, &apiObject.PrimaryRegion)...)
+	diags.Append(flex.Expand(ctx, m.RecoveryTimeObjectiveMinutes, &apiObject.RecoveryTimeObjectiveMinutes)...)
+	diags.Append(flex.Expand(ctx, m.Triggers, &apiObject.Triggers)...)
+	diags.Append(flex.Expand(ctx, m.Tags, &apiObject.Tags)...)
+
+	// Handle AssociatedAlarms set-to-map conversion manually
```

Comment:
The `AssociatedAlarms` struct is relatively simple - can you elaborate on why AutoFlex doesn't handle this with its default behavior?